lib/vanagon/component.rb

```diff
@@ -142,14 +144,18 @@ def get_build_dir
     # Fetches secondary sources for the component. These are just dumped into the workdir currently.
     #
     # @param workdir [String] working directory to put the source into
-    def get_sources(workdir)
+    def get_sources(workdir) # rubocop:disable Metrics/AbcSize
+      # create an empty array if @extract_with doesn't already exist
+      @extract_with = Array(@extract_with)
       sources.each do |source|
         src = Vanagon::Component::Source.source source.url,
-                                                workdir: workdir,
-                                                ref: source.ref,
-                                                sum: source.sum
+                                                  workdir: workdir,
+                                                  ref: source.ref,
+                                                  sum: source.sum
         src.fetch
         src.verify
+        @file = File.basename(src.url)
```

Comment:
might be good to add a comment about how this is needed to make `extract()` behave?